pkg/apis/kubeone/v1beta1/types.go

```diff
@@ -590,12 +590,31 @@ type OpenIDConnectConfig struct {
 	CAFile string `json:"caFile"`
 }
 
+// Addon config
+type Addon struct {
+	// Name of the addon to configure
+	Name string `json:"name"`
+
+	// Params to the addon, to render the addon using text/template, this will override globalParams
+	Params map[string]string `json:"params,omitempty"`
+
+	// Delete flag to ensure the named addon with all its contents to be deleted
+	Delete bool `json:"deleted,omitempty"`
+}
+
 // Addons config
 type Addons struct {
 	// Enable
 	Enable bool `json:"enable,omitempty"`
+
 	// Path on the local file system to the directory with addons manifests.
-	Path string `json:"path"`
+	Path string `json:"path,omitempty"`
+
+	// GlobalParams to the addon, to render all addons using text/template
+	GlobalParams map[string]string `json:"globalParams,omitempty"`
+
+	// Addons is a list of config options for named addon
+	Addons []Addon `json:"addons,omitempty"`
```

Comment:
Have you implemented enabling embedded addons via this API yet?

Reply:
No, this is out of scope.

Reply:
This is actually the key part of the Epic. Passing parameters to addons was supposed to be only a nice addition if we have time left, but not the key part. Quoting part of the Epic description:

> We need to implement an API that can be used to enable optional addons embedded in the binary, such as `cluster-autoscaler` and `unattended-upgrades`.

Can you please extend the PR to handle this as well? I think it should be fairly easy, but it might take some time to test it properly.